### ISSUE TITLE
Xerces 3.2.0 is not supported

### ISSUE THREAD

**Issue by momeara** [OpenMS/OpenMS#3073]
On the [Building OpenMS on GNU/Linux](http://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/html/install_linux.html) page, it says OpenMS requires

    Xerces-C (>= 3.1.1)

But Xerces-C version `3.2.0` is not supported:

```
cd ~/opt
wget http://apache.forsale.plus//xerces/c/3/sources/xerces-c-3.2.0.tar.gz
tar -xzvf xerces-c-3.2.0.tar.gz
mkdir -p xerces-c-3.2.0-build
cd xerces-c-3.2.0-build
cmake -DCMAKE_INSTALL_PREFIX=~/opt ../xerces-c-3.2.0
make
make install
```

```
cd ~/opt
git clone git@github.com:OpenMS/OpenMS.git
mkdir -p OpenMS-build
cd OpenMS-build
cmake \
    -DOPENMS_CONTRIB_LIBS="$BUILD_BASE/contrib-build" \
    -DBOOST_USE_STATIC=OFF \
    -DWITH_GUI=OFF \
    -DHAS_XSERVER=OFF \
    -DXercesC_LIBRARY=~/opt/lib64/libxerces-c.so \
    ../OpenMS
make
make install
```

This fails with 

```
[ 30%] Linking CXX executable ../../bin/PrecursorMassCorrector
CMakeFiles/IDRipper.dir/IDRipper.cpp.o: In function `xercesc_3_1::DefaultHandler::fatalError(xercesc_3_1::SAXParseException const&)':
IDRipper.cpp:(.text._ZN11xercesc_3_114DefaultHandler10fatalErrorERKNS_17SAXParseExceptionE[_ZN11xercesc_3_114DefaultHandler10fatalErrorERKNS_17SAXParseExceptionE]+0x1d): undefined reference to `xercesc_3_1::SAXParseException::SAXParseException(xercesc_3_1::SAXParseException const&)'
IDRipper.cpp:(.text._ZN11xercesc_3_114DefaultHandler10fatalErrorERKNS_17SAXParseExceptionE[_ZN11xercesc_3_114DefaultHandler10fatalErrorERKNS_17SAXParseExceptionE]+0x24): undefined reference to `xercesc_3_1::SAXParseException::~SAXParseException()'
IDRipper.cpp:(.text._ZN11xercesc_3_114DefaultHandler10fatalErrorERKNS_17SAXParseExceptionE[_ZN11xercesc_3_114DefaultHandler10fatalErrorERKNS_17SAXParseExceptionE]+0x2b): undefined reference to `typeinfo \
for xercesc_3_1::SAXParseException'
../../lib/libOpenMS.so: undefined reference to `xercesc_3_1::InputSource::setEncoding(unsigned short const*)'
```

**Comment by hroest** [comment]
thanks for reporting this, it seems like this version 3.2.0 was just released in August 2017 while our release (and the documentation you link to) was done in July 2017. So until we have this sorted out, I would suggest to use older xerces versions. 

@timosachsenberg @jpfeuffer we should change the documentation of the new OpenMS release to reflect this.

**Comment by hroest** [comment]
hmm even though xerces says its API compatible: https://xerces.apache.org/xerces-c/migrate-3.html

* Xerces-C++ 3.2.0 is an API-compatible, but not ABI-compatible, update to the 3.x branch. Code designed for use with Xerces 3 should continue to compile, but existing applications MUST be recompiled to work with this version.

**Comment by momeara** [comment]
It looks like it may be a namespacing issue from the error message not
necessarily a functionality issue.

I can confirm that it no longer gives that error with version 3.1.0, which
is available here:

https://archive.apache.org/dist/xerces/c/3/sources/xerces-c-3.1.0.tar.gz

**Comment by hroest** [comment]
I have just build openms using the most recent Ubuntu 18.04 which has xerces 3.2.0 and this seems to work well:

```
-- Found XercesC: /usr/lib/x86_64-linux-gnu/libxerces-c.so (found version "3.2.0") 
...
$ make -j6 && echo "done"
...
[100%] Built target Tutorials_build
[100%] Built target extra_targets
done
```

so it does not seem to be an issue when building with the distribution provided xerces library.

**Comment by cbielow** [comment]
Could this be an include-path issue?! I.e. if xerces headers from the contrib are used during compiling, but the 3.2 lib from the user?

**Comment by hroest** [comment]
seems the most likely scenario since it works when having only xerces 3.2 - this is possibly related to #2558 and #2562 - but then it should have been solved by #2623 right?
